include/wil/registry.h

```diff
@@ -162,6 +162,108 @@ namespace wil
         }
 #endif // #define __WIL_WINREG_STL
 
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        //
+        // wil::reg::value_enumerator<HKEY> enumerate_values(HKEY key)
+        // wil::reg::key_enumerator<HKEY> enumerate_keys(HKEY key)
+        //
+        //  - Enumerates registry keys and values under the given HKEY
+        //  - Throws a std::exception on failure (including wil::ResultException)
+        //
+        // Examples of usage when std::wstring is included:
+        //
+        //     for (const auto& key_data : wil::make_range(wil::reg::key_iterator{hkey}, wil::reg::key_iterator{}))
+        //     {
+        //         key_data.name; // the std::wstring of the enumerated key
+        //     }
+        //
+        //     for (const auto& key_data : wil::reg::enumerate_keys(hkey))
+        //     {
+        //         key_data.name; // the std::wstring of the enumerated key
+        //     }
+        //
+        //     for (const auto& value_data : wil::make_range(wil::reg::value_iterator{hkey}, wil::reg::value_iterator{}))
+        //     {
+        //         value_data.name; // the std::wstring of the enumerated value
+        //         value_data.type; // the REG_ type of the enumerated value
+        //     }
+        //
+        //     for (const auto& value_data : wil::reg::enumerate_values(hkey.get()))
+        //     {
+        //         value_data.name; // the std::wstring of the enumerated value
+        //         value_data.type; // the REG_ type of the enumerated value
+        //     }
+        //
+        // When std::wstring is *not* included, the above functions return a 'name' field of type wil::unique_process_heap_string
+        //
+        /**
+         * \brief Returns an enumerator object that exposes begin() and end() to iterate through the registry values under the specified key
+         * \param key An open registry key
+         *        Note: the caller must guarantee the registry key must be valid for the lifetime of the returned registry enumerator object
+         * \return wil::reg::value_enumerator The object to iterate values - exposing the iterator semantics of being() and end()
+         */
+        inline ::wil::reg::value_enumerator<HKEY> enumerate_values(HKEY key)

```

Comment:
When I see a function named `enumerate*` I expect it to take a callback as a function argument, not return an iterator. Using a name like `make_` or `create_` would alleviate such confusion